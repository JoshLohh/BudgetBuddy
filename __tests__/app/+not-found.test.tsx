import React from 'react';
import { render } from '@testing-library/react-native';
import NotFoundScreen from '../../app/+not-found.tsx';
console.log('NotFoundScreen:', NotFoundScreen);

jest.mock('@/components/ThemedText', () => {
  const React = require('react');
  const { Text } = require('react-native');
  return {
    ThemedText: ({ children, ...props }: any) => <Text {...props}>{children}</Text>,
  };
});

jest.mock('@/components/ThemedView', () => {
  const React = require('react');
  return {
    ThemedView: ({ children, ...props }: any) => <>{children}</>,
  };
});

jest.mock('expo-router', () => {
    const React = require('react');
    const { Text, View } = require('react-native');
    // First, create the Stack function
    const Stack = ({ children } : any) => <>{children}</>;
    // Now, assign a Screen static property (can be a placeholder)
    Stack.Screen = ({ children }: any) => <>{children}</>;
    return {
      Link: ({ children, ...props }: any) => <Text accessibilityRole="link" {...props}>{children}</Text>,
      Stack,
    };
});

describe('+not-found.tsx', () => {
  it('renders the not found message', () => {
    const { getByText } = render(<NotFoundScreen />);
    expect(getByText('This screen does not exist.')).toBeTruthy();
  });

  it('renders the link to go to home screen', () => {
    const { getByText } = render(<NotFoundScreen />);
    expect(getByText('Go to home screen!')).toBeTruthy();
  });

  it('renders within a themed view (styles are applied)', () => {
    const { getByText } = render(<NotFoundScreen />);
    expect(getByText('This screen does not exist.')).toBeTruthy();
  });

  it('link to home screen is accessible and behaves correctly', () => {
    const { getByRole } = render(<NotFoundScreen />);
    const link = getByRole('link');
    expect(link).toBeTruthy();
  });
  
  // Snapshot test added here
  it('matches the snapshot', () => {
    const { toJSON } = render(<NotFoundScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
