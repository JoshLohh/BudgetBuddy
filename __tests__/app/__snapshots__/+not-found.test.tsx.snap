// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+not-found.tsx matches the snapshot 1`] = `
[
  <Text
    type="title"
  >
    This screen does not exist.
  </Text>,
  <Text
    accessibilityRole="link"
    href="/"
    style={
      {
        "marginTop": 15,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      type="link"
    >
      Go to home screen!
    </Text>
  </Text>,
]
`;
